source/s3_copy_object.c

```diff
@@ -7,6 +7,7 @@
 #include "aws/s3/private/s3_request_messages.h"
 #include "aws/s3/private/s3_util.h"
 #include <aws/common/string.h>
+#include <aws/http/private/strutil.h>
```

Comment:
this is a layering violation: private headers can only be used inside the module that declares them. We should refactor this so it can be more generic (possibly in aws-c-common)

Reply:
Created a task to move `aws_strutil_read_unsigned_num` to `aws-c-common` to enable reuse between `aws-c-http` and `aws-c-s3`. The refactor will be implemented in distinct PRs.